analyzer/src/main/kotlin/managers/utils/DependencyGraphConverter.kt

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.analyzer.managers.utils
+
+import org.ossreviewtoolkit.model.AnalyzerResult
+import org.ossreviewtoolkit.model.DependencyGraph
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.OrtIssue
+import org.ossreviewtoolkit.model.Package
+import org.ossreviewtoolkit.model.PackageLinkage
+import org.ossreviewtoolkit.model.PackageReference
+import org.ossreviewtoolkit.model.Project
+
+/**
+ * An object that supports the conversion of [AnalyzerResult]s to the dependency graph format.
+ *
+ * The main conversion function takes an [AnalyzerResult] and iterates over all projects contained in it. Each
+ * project that still uses the dependency tree representation (i.e. the set of scopes) is converted, and a dependency
+ * graph for the associated package manager is created.
+ *
+ * Via this converter, optimized results can be generated for package managers that do not yet support this format.
+ * This approach is, however, not ideal, because during the conversion both the dependency tree and the dependency
+ * graph need to be hold in memory. So this is rather a temporary solution until all package managers support the new
+ * format natively.
+ */
+object DependencyGraphConverter {
```

Comment:
Is there any reason that this class has to be in the analyzer module? I think it could be in `model.utils` as well which would make the reverse dependency you introduce in the next commits unnecessary. This would require moving `DependencyGraphBuilder` and maybe some other classes to the model as well, but I think I would prefer that over the service loader approach. As this is an architectural question I'd like to have another opinion from @sschuberth or @fviernau as well.

Reply:
I'm not a fan of Java's `ServiceLoader` mechanism in general, so whenever we can (reasonably) avoid it, I'm in for it. And simply moving (hopefully only) a few classes to the `model` module to do so sounds reasonable to me.

Reply:
I was not sure about this, so I was hoping for your feedback.

So I take it, the agreement is to move the `DependencyGraphBuilder` and related classes to `model.utils`. Then the custom serializer for `AnalyzerResult` can be configured in the usual way via annotations, and no manipulations on standard mappers are needed.

Reply:
The following classes are now under `model.utils`:
* `DependencyGraphBuilder`
* `DependencyHandler`
* `DependencyGraphConverter`